### PR TITLE
Remove deprecated/not-needed `hapi` types

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by codershiba** [comment]
@jakebailey Is thik ok or should I split it into several PRs?

File: notNeededPackages.json [hunk 1]
```diff
@@ -904,6 +912,10 @@
             "libraryName": "cnpj",
             "asOfVersion": "3.1.0"
         },
+        "code": {
```
Comment (jakebailey) [inline notNeededPackages.json:915]: Most of these are missing the hapi prefix and are going to touch the wrong packages.

**Review by jakebailey — DISMISSED** [review]
Unless the packages are interrelated, separate PRs are usually easier to manage. But this PR currently does not look correct.

File: notNeededPackages.json (hunk 1 above)
Comment (codershiba) [reply notNeededPackages.json:915]: Is this actually wrong?
https://www.npmjs.com/package/@types/code seems to provide types for https://github.com/hapijs/code which now seems to ship its own types.

Comment (jakebailey) [reply notNeededPackages.json:915]: Hm, maybe not; the bot doesn't seem to have noted that `code` was being deleted, which is what I was going off of:

image

Clearly I didn't scroll down enough and trusted the bot to be accurate.

Comment (jakebailey) [reply notNeededPackages.json:915]: Oh, you marked it as a draft, so the bot stopped looking at it.

**Comment by jakebailey** [comment]
So generally speaking, I'm apprehensive about replacing `@types/boom` with `@hapi/boom` (and for all of the other ones); `@types/boom` are the types for the `boom` package. What this will mean is that people will install `boom` (say, in an old project), then install `@types/boom`, which will then via the replacement shim install `@hapi/boom`. But then, still nothing is providing types for `boom`, which the user requested! `boom` still gets 1.5M downloads and its types get 50k downloads, and this change will break those users.

Realistically, the problem is that these hapi packages were deprecated and moved somewhere else. If we're saying that we don't want them on DT anymore and they aren't going to change, the only two options I see are to either delete them without modifying `notNeededPackages.json`, or to just continue to leave them.

**Comment by codershiba** [comment]
> So generally speaking, I'm apprehensive about replacing `@types/boom` with `@hapi/boom` (and for all of the other ones); `@types/boom` are the types for the `boom` package. What this will mean is that people will install `boom` (say, in an old project), then install `@types/boom`, which will then via the replacement shim install `@hapi/boom`. But then, still nothing is providing types for `boom`, which the user requested! `boom` still gets 1.5M downloads and its types get 50k downloads, and this change will break those users. (For reference, adding to `notNeededPackage.json` publishes packages containing only https://github.com/microsoft/DefinitelyTyped-tools/blob/main/packages/publisher/src/generate-packages.ts#L155.)
> 
> Realistically, the problem is that these hapi packages were deprecated and moved somewhere else. If we're saying that we don't want them on DT anymore and they aren't going to change, the only two options I see are to either delete them without modifying `notNeededPackages.json`, or to just continue to leave them.

I think we should remove them without modifying `notNeededPackages.json` as most of these packages have not been updated in the last few years and just seems like a maintenance burden.

**Comment by jakebailey** [comment]
Yes. So I'd update this PR to reflect that.

**Pull request closed without merging**